File: AtCoder/abc206/C.swift
private func readLn() -> String {
	return readLine() ?? " "
}

private func readLns() -> [String] {
	return readLn().split(separator: " ").map{ String($0) }
}

private func readInt() -> Int {
	return Int(readLn()) ?? 0
}

private func readInts() -> [Int] {
	return readLns().compactMap{ Int($0) }
}

let n = readInt()
let arr = readInts()
var mp = [Int : Int]()
var ans = Int64(0)
for (i, ele) in arr.enumerated() {
	ans += Int64(i - (mp[ele] ?? 0))
	mp[ele] = (mp[ele] ?? 0) + 1
}
print(ans)

File: AtCoder/abc206/C.kt
private fun readLn() = readLine()!!
private fun readLns() = readLn().split(" ")
private fun readInt() = readLn().toInt()
private fun readInts() = readLns().map { it.toInt() }

fun main() {
    val n = readInt()
    val arr = readInts()
    var ans = 0L
    val mp = hashMapOf<Int, Int>()
    for (i in arr.indices) {
        val ele = arr[i]
        ans += i - mp.getOrDefault(ele, 0)
        mp[ele] = mp.getOrDefault(ele, 0) + 1
    }
    print(ans)
}